docs/chainlink-keepers/flexible-upkeeps.md

```diff
@@ -0,0 +1,157 @@
+---
+layout: nodes.liquid
+section: ethereum
+date: Last Modified
+title: 'Build highly flexible, cheap, and secure automation using Chainlink Keepers'
+whatsnext: { 'Utility Contracts': '/docs/chainlink-keepers/util-overview/', 'FAQs': '/docs/chainlink-keepers/faqs/' }
+---
+
+## Overview
+
+In this tutorial, you will learn how the flexibility of [Chainlink Keepers](https://chain.link/keepers) enables important design patterns that reduce gas fees, enhance the resilience of your dApps, and improve your end-user experience.
+In the following sections, you will start by integrating an unoptimized contract to Chainlink Keepers (cf. [Problem](#problem-on-chain-computation-leads-to-high-gas-fees)). Then, you will see how properly leveraging the flexibility of Chainlink Keepers (cf. [Solution section](#solution-perform-complex-computations-with-no-gas-fees)) allows you to save ~84% of gas fees!
+
+**Table of Contents**
+
+- [Prerequisites](#prerequisites)
+- [Problem: On-chain computation leads to high gas fees](#problem-on-chain-computation-leads-to-high-gas-fees)
+  - [Problem statement](#problem-statement)
+  - [Example](#example)
+- [Solution: Perform complex computations with no gas fees](#solution-perform-complex-computations-with-no-gas-fees)
+- [Conclusion](#conclusion)
+
+## Prerequisites
+
+Smart contracts cannot self-trigger their functions at arbitrary times or under arbitrary conditions. Transactions can only be initiated by another account.
+On August 5th, 2021, we were very excited to [announce](https://blog.chain.link/chainlink-keepers-is-now-live-on-mainnet/) an important [hybrid smart contract](https://blog.chain.link/hybrid-smart-contracts-explained/) innovation: [Chainlink Keepers](https://chain.link/keepers).
+By using the decentralized solution implemented by Chainlink Keepers, developers don’t need to rely on centralized services or implement the infrastructure themselves to automate on-chain functions.
+Several Defi projects are already live on [Ethereum Mainnet](https://keepers.chain.link/mainnet), [Polygon Mainnet](https://keepers.chain.link/polygon) & [Binance Smart Chain Mainnet](https://keepers.chain.link/bsc) and benefit from the decentralized and secure automation brought by Chainlink Keepers.
+
+This tutorial assumes you have a basic understanding of [Chainlink Keepers](https://chain.link/keepers). If you are new to this concept then please start with the following:
+
+- [Chainlink Keepers announcement](https://blog.chain.link/chainlink-keepers-is-now-live-on-mainnet/)
```

Comment:
Which part should I remove: the blog announcement or the whole prerequisites section? Because this is an advanced tutorial, one should be first familiar with the basics (how to make a keepers compatible contract..ETc)